dockstore-webservice/src/main/java/io/dockstore/webservice/resources/DockerRepoResource.java

```diff
@@ -816,6 +817,7 @@ public List<Tool> allPublishedContainers(
     public List<Tool> getPublishedContainerByPath(
         @ApiParam(value = "repository path", required = true) @PathParam("repository") String path) {
         List<Tool> tools = toolDAO.findAllByPath(path, true);
+        checkNotNull(tools, "Invalid repository path");
```

Comment:
Does the previous line not return an empty list instead of null if nothing matches?

Reply:
`findAllByPath` returns `null` if the path is invalid, and otherwise, a list of what matches (which could be empty if nothing matched).